lib/std/target/aarch64.zig

```diff
@@ -0,0 +1,1603 @@
+        &feature_rcpc,
+        &feature_uaops,
+        &feature_rdm,
+        &feature_ras,
+        &feature_lse,
+        &feature_crc,
+        &feature_perfmon,
+        &feature_fpArmv8,
+        &feature_vh,
+        &feature_fuseAes,
+        &feature_lor,
+        &feature_dotprod,
+        &feature_pan,
+    },
+};
+
+pub const cpu_cortexA76 = Cpu{
+    .name = "cortexA76",
+    .llvm_name = "cortex-a76",
+    .dependencies = &[_]*const Feature {
+        &feature_ccpp,
+        &feature_rcpc,
+        &feature_uaops,
+        &feature_rdm,
+        &feature_ras,
+        &feature_lse,
+        &feature_crc,
+        &feature_fpArmv8,
+        &feature_vh,
+        &feature_lor,
+        &feature_ssbs,
+        &feature_dotprod,
+        &feature_pan,
+    },
+};
+
+pub const cpu_cortexA76ae = Cpu{
+    .name = "cortexA76ae",
+    .llvm_name = "cortex-a76ae",
+    .dependencies = &[_]*const Feature {
+        &feature_ccpp,
+        &feature_rcpc,
+        &feature_uaops,
+        &feature_rdm,
+        &feature_ras,
+        &feature_lse,
+        &feature_crc,
+        &feature_fpArmv8,
+        &feature_vh,
+        &feature_lor,
+        &feature_ssbs,
+        &feature_dotprod,
+        &feature_pan,
+    },
+};
+
+pub const cpu_cyclone = Cpu{
+    .name = "cyclone",
+    .llvm_name = "cyclone",
+    .dependencies = &[_]*const Feature {
+        &feature_arithCbzFusion,
+        &feature_zczFpWorkaround,
+        &feature_alternateSextloadCvtF32Pattern,
+        &feature_fuseCryptoEor,
+        &feature_zcm,
+        &feature_zczGp,
+        &feature_perfmon,
+        &feature_disableLatencySchedHeuristic,
+        &feature_fpArmv8,
+        &feature_zczFp,
+        &feature_arithBccFusion,
+        &feature_fuseAes,
+    },
+};
+
+pub const cpu_exynosM1 = Cpu{
+    .name = "exynosM1",
+    .llvm_name = "exynos-m1",
+    .dependencies = &[_]*const Feature {
+        &feature_customCheapAsMove,
+        &feature_crc,
+        &feature_force32bitJumpTables,
+        &feature_perfmon,
+        &feature_slowMisaligned128store,
+        &feature_useReciprocalSquareRoot,
+        &feature_fpArmv8,
+        &feature_zczFp,
+        &feature_fuseAes,
+        &feature_slowPaired128,
+        &feature_usePostraScheduler,
+    },
+};
+
+pub const cpu_exynosM2 = Cpu{
+    .name = "exynosM2",
+    .llvm_name = "exynos-m2",
+    .dependencies = &[_]*const Feature {
+        &feature_customCheapAsMove,
+        &feature_crc,
+        &feature_force32bitJumpTables,
+        &feature_perfmon,
+        &feature_slowMisaligned128store,
+        &feature_fpArmv8,
+        &feature_zczFp,
+        &feature_fuseAes,
+        &feature_slowPaired128,
+        &feature_usePostraScheduler,
+    },
+};
+
+pub const cpu_exynosM3 = Cpu{
+    .name = "exynosM3",
+    .llvm_name = "exynos-m3",
+    .dependencies = &[_]*const Feature {
+        &feature_fuseLiterals,
+        &feature_predictableSelectExpensive,
+        &feature_customCheapAsMove,
```

Comment:
according to

https://github.com/llvm/llvm-project/blob/release/9.x/llvm/lib/Target/AArch64/AArch64.td#L582

this should be FeatureExynosCheapAsMoveHandling.

how did these lists get generated? looks like the data needs to be audited.

Reply:
Crap, looks like you found a bug in my generation scripts... sorry about that. I took a look and it appears that there is an issue with features with dependencies not being included in the CPU feature list (but their dependencies make it). I fixed it and manually verified that this CPU is correct. Would you like me to push the updated CPU lists?

Reply:
Generating these is a pain. I run LLVM's tablegen on each arch's `.td` and parse the features/CPUs in there and resolve their feature bitmaps to get the final list.

Reply:
is your project open source that generates the data from the .tds? it might be nice to run it after every llvm release and look at a diff.

I went with a different organization, if you look at this branch: https://github.com/ziglang/zig/tree/layneson-cpus_and_features

The aarch64.zig file is updated to the refactored organization. If you were willing to adjust your scripts and run it according to this new data layout, that would save me a lot of time

Reply:
The main idea here was using a bit set to represent all the CPU features, rather than a slice of pointers to structs. This allows std.Target to maintain its current property that it does not need to be heap allocated no matter what the list of cpu features is.

Reply:
Didn't realize until a few minutes ago (when I saw the livestream VOD) that the data layout was changed. I will clean up my scripts, modify them to output the new data layout, and put that on GitHub. It would be nice to have a way of verifying the final output, but I'm not immediately sure of a way to do that without repeating any mistakes/wrong assumptions from the original generation code.

One question: how did you generate the tag names for the `Feature` enums? Did you go from the LLVM name and replace `-` with `_`? What about other characters (such as spaces and `+`)?

Reply:
heads up, I'm about to rebase this branch against master

Reply:
> One question: how did you generate the tag names for the Feature enums? Did you go from the LLVM name and replace - with _? What about other characters (such as spaces and +)?

Yeah it's the LLVM name, replacing - with _ (following the [style guide](https://ziglang.org/documentation/master/#Style-Guide)). When this would lead to an identifier being illegal, put it in `@"foo"` (string literal identifier syntax).

Reply:
@andrewrk I rewrote my generation scripts. The project can now be found [here](https://github.com/layneson/zig-llvm-target-details-generator). The output looks correct (to a few minutes of hand-verification), but even diffing your <arch>.zig files against my generated ones doesn't work since the ordering of feature and CPU defs differs between the two, so I am not certain that it is correct. Let me know if you'd like any changes / something doesn't look right!

Reply:
Thanks! I'll try this out now.